src/routes/extras/containers/ExtrasContainer.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import * as React from 'react'
+import connect from 'react-redux/es/connect/connect'
+import { Linking, ActivityIndicator } from 'react-native'
+import Extras from '../components/Extras'
+import { TFunction, translate } from 'react-i18next'
+import compose from 'lodash/fp/compose'
+import { createExtrasEndpoint, ExtraModel, Payload } from '@integreat-app/integreat-api-client'
+import type { ThemeType } from '../../../modules/theme/constants/theme'
+import request from '../../../modules/endpoint/request'
+import type { StateType } from '../../../modules/app/StateType'
+import type { NavigationScreenProp } from 'react-navigation'
+import { baseUrl } from '../../../modules/endpoint/constants'
+
+const mapStateToProps = (state: StateType, ownProps) => {
+  const language = state.cityContent.language
+  if (!language) {
+    throw new Error('Language not set for categories!')
+  }
+
+  const targetCity: string = ownProps.navigation.getParam('cityCode')
+
+  return {
+    city: targetCity,
+    language: language
+  }
+}
+
+type PropsType = {|
+  navigation: NavigationScreenProp<*>,
+  city: string,
+  language: string,
+  navigateToExtra: (path: string, isExternalUrl: boolean) => void,
+  theme: ThemeType,
+  t: TFunction
+|}
+
+type ExtrasStateType = {|
+  extras: ?Array<ExtraModel>,
+  error: ?Error
+|}
+
+class ExtrasContainer extends React.Component<PropsType, ExtrasStateType> {
+  constructor (props: PropsType) {
+    super(props)
+    this.state = {extras: null, error: null}
+  }
+
+  componentWillMount () {
+    this.loadExtras()
+  }
+
+  navigateToExtra = (path: string, isExternalUrl: boolean, offerHash: ?string = null) => {
+    if (isExternalUrl) {
+      Linking.openURL(path)
+    } else if (this.props.navigation.push) {
+      const params = {city: this.props.city, extras: this.state.extras, offerHash: offerHash}
+      this.props.navigation.push(path, params)
+    }
+  }
+
+  async loadExtras () {
```

Comment:
Is it on purpose to have this loading/fetching code here? This isn't handled the same way for events/categories, is it?

Reply:
Extras are only available when you are online. That's why I decided not to put it in the store. Very simple to do it like this.